### ISSUE TITLE
[iOS] Native crash:  NSRangeException: *** -[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty NSArray

### ISSUE THREAD

**Issue by max-kim** [mateusz1913/react-native-avoid-softinput#131]
### Environment

Library version: 3.0.2
OS version: iOS 14-16

### Affected platforms

- [ ] Android
- [X] iOS

### Current behavior

Facing native crash on prod env. Happens when it tries to get the root view, the [array of subviews is empty](https://github.com/mateusz1913/react-native-avoid-softinput/blob/b9eced9be69c1fd5185838b3dae571892ed378b5/packages/react-native-avoid-softinput/ios/AvoidSoftInputObjCPPUtils.mm#L30) for some reason. What is curious it happens only once per user. 
```
NSRangeException: *** -[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty NSArray
  File "AvoidSoftInputObjCPPUtils.mm", line 30, in +[AvoidSoftInputObjCPPUtils getReactRootViewWithRootViewController:]
  File "AvoidSoftInputManager.swift", line 107, in AvoidSoftInputManager.softInputHeightWillChange
  File "AvoidSoftInputImpl.swift", line 151, in AvoidSoftInputImpl.softInputHeightWillChange
  File "<compiler-generated>", in AvoidSoftInputImpl.softInputHeightWillChange
  File "main.m", line 7, in main
```

Unfortunately can't reproduce it on dev env.

### Expected behavior

No crashes because of the reason.
I've made a patch with an additional condition to check if the array of subviews is empty and am wondering if it's suitable for you as a general solution.
```
{
  if ([
        viewController.view isKindOfClass:[
#ifdef RCT_NEW_ARCH_ENABLED
        RCTFabricSurfaceHostingProxyRootView
#else
        RCTRootView
#endif
        class]] &&  viewController.view.subviews.count > 0)
    {
    return viewController.view.subviews[0];
  }

  return viewController.view;
}
```

### Reproduction

-

**Comment by mateusz1913** [comment]
Hey @max-kim , thx for reporting the issue! Could you make a PR with that patch? I would be very happy to merge it

**Comment by mateusz1913** [comment]
Thx so much 🎉 the fix is included in version [3.0.4](https://github.com/mateusz1913/react-native-avoid-softinput/releases/tag/v3.0.4)
